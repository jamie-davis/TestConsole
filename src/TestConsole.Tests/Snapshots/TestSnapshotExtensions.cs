using System;
using System.Collections.Generic;
using System.Text;
using NUnit.Framework;
using TestConsoleLib;
using TestConsoleLib.Snapshots;
using TestConsoleLib.Testing;

namespace TestConsole.Tests.Snapshots
{
    [TestFixture]
    public class TestSnapshotExtensions
    {
        private SnapshotCollection _collection;

        [SetUp]
        public void SetUp()
        {
            _collection = new SnapshotCollection();
        }

        [Test]
        public void CollectionsCanBePlacedInSnapshots()
        {
            //Arrange
            var data = new[]
            {
                new {A = 1, B = "Bee", C = 5.5},
                new {A = 2, B = "Cee", C = 6.6},
                new {A = 3, B = "Dee", C = 7.7},
                new {A = 4, B = "Eee", C = 8.8},
                new {A = 5, B = "Eff", C = 9.9},
            };

            var snapshot = _collection.NewSnapshot("Test");

            //Act
            data.ToSnapshotTable(snapshot, "Data");

            //Assert
            var output = new Output();
            _collection.GetSnapshot("Test").ReportContents(output);
            output.Report.Verify();
        }

        [Test]
        public void TablesCanBeAutomaticallyDefined()
        {
            //Arrange
            var data = new[]
            {
                new {A = 1, B = "Bee", C = 5.5},
                new {A = 2, B = "Cee", C = 6.6},
                new {A = 3, B = "Dee", C = 7.7},
                new {A = 4, B = "Eee", C = 8.8},
                new {A = 5, B = "Eff", C = 9.9},
            };

            var snapshot = _collection.NewSnapshot("Test");

            //Act
            data.ToSnapshotTable(snapshot, "Data", "A", "B");

            //Assert
            var output = new Output();
            var table = _collection.GetTableDefinition("Data");
            _collection.GetSchemaReport(output);
            output.Report.Verify();
        }
    }
}
