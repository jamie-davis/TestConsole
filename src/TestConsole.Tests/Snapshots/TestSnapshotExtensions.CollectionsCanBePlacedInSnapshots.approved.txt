Data                                                                                           
----                                                                                           
           A B                       C                                                         
------------ ------------ ------------                                                         
           1 Bee                  5.50                                                         
           2 Cee                  6.60                                                         
           3 Dee                  7.70                                                         
           4 Eee                  8.80                                                         
           5 Eff                  9.90                                                         
                                                                                               